dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -18,21 +20,26 @@ function DatePickerWidget({
   setDateRange,
 }) {
   const [fromDate, setFromDate] = useState({});
-  const [strDate, setStrDate] = useState(formatDate(new Date()));
-  const [toDate, setToDate] = useState(constructUTCDate(new Date(strDate)));
+  const [toDate, setToDate] = useState(
+    constructUTCDate(formatDate(new Date(bumpToDate(new Date(), 1))))
+  );
+  const [strDate, setStrDate] = useState(
+    new Date().toLocaleString("fr-CA").split(",")[0]
+  );
   const toValidator = (date) =>
     date >= fromDate
       ? ""
       : "To date must be greater than or equal to from date";
 
   const onFromChange = (_str, date) => {
-    setFromDate(isValidDate(date) ? constructUTCDate(new Date(_str)) : {});
+    const selectedDate = constructUTCDate(_str);
+    setFromDate(isValidDate(date) ? selectedDate : {});
     if (isValidDate(date)) {
-      if (date > toDate) {
-        let selectedDate = new Date(_str);
-        selectedDate.setDate(selectedDate.getDate() + 1);
-        setToDate(constructUTCDate(selectedDate));
-        setStrDate(formatDate(selectedDate));
+      if (date > new Date(strDate)) {
+        setToDate(
+          constructUTCDate(formatDate(new Date(bumpToDate(new Date(_str), 2))))
+        );
```

Comment:
My first question here is, why are you bumping the date forward by _two_ days?  (If you stick with that, it deserves a comment!)

It strikes me, lacking any better value for the to-date, that what we want is for the from-date to be the beginning of the day and the to-date to be the end of the same day.  Since we apparently have some concerns about using 24:00 as a time, that means that the to-date should be one full day later than the from-date...but only **one** day.  (If you think you need _two_, then I think there's a problem somewhere else...but I could be wrong...it's happened at least once in this PR already.  😉)

Next, I'd like to propose an alternate flow, here.  Given what I've said here, once execution reaches line 39, we have a value in `selectedDate` that we like for `fromDate`, but it's a day behind where we want it to be for `toDate`.  So, rather than starting all over with `_str`, lets continue on from `selectedDate`:
```
        setToDate(bumpToDate(new Date(selectedDate), 1));
```
That is, create a new `Date` with the same value as `selectedDate`, bump it forward, and use the result for `toDate`.  (Now we're creating only one `Date` object instead of a sequence of three.)